pkg/api/api_test.go

```diff
@@ -47,201 +47,183 @@ type testCases struct {
 	optionsTransportCalled       bool
 	multipartFormTransportCalled bool
 	websocketTransportCalled     bool
+	init                         func(e *echo.Echo, tt testStartServerType) *gomonkey.Patches
 }
 
-func TestStart(t *testing.T) {
-	tests := initializeTestCases()
-
-	patches := applyPatches()
-	defer patches.Reset()
-
-	mockEchoServer, _ := prepareMocks()
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			setupMockFunctions()
-
-			if hasTransportCalls(tt) {
-				checkTransportCalls(tt)
-			} else {
-				checkGraphQLServer(t, tt, mockEchoServer)
-			}
-		})
+func testStartServerSuccessCase() testStartServerType {
+	return testStartServerType{
+		name: "Success",
+		args: args{
+			cfg: testutls.MockConfig(),
+		},
+		setDbCalled: true,
+		wantErr:     false,
+		init: func(e *echo.Echo, tt testStartServerType) *gomonkey.Patches {
+			return gomonkey.ApplyFunc(os.Getenv, func(key string) (value string) {
+				if key == "JWT_SECRET" {
```

Comment:
let's do something like this instead 

https://github.com/wednesday-solutions/go-template/blob/feat/linters-patch/internal/config/env_test.go#L195-L207